server/game/GameActions/SwapEverythingAction.js

```diff
@@ -0,0 +1,51 @@
+const CardGameAction = require('./CardGameAction');
+
+class SwapEverythingAction extends CardGameAction {
```

Comment:
Not sure I like the name, though.
What do you think about adding a pair of attributes in the SwapAction: `swapTokens = false`, `swapUpgrades = false`?

Reply:
Good idea, done!  Please take another look.

Reply:
Beautiful